iothub/device/tests/Pipeline/RetryDelegatingHandlerTests.cs

```diff
@@ -221,14 +221,18 @@ public async Task DeviceNotFoundExceptionReturnsDeviceDisabledStatus()
             var sut = new RetryDelegatingHandler(contextMock, nextHandlerMock);
 
             // act
-            await ((Func<Task>)(() => sut
-                .OpenAsync(CancellationToken.None)))
-                .ExpectedAsync<IotHubClientException>()
-                .ConfigureAwait(false);
+            Func<Task> act = async () =>
+            {
+                await sut
+                .OpenAsync(CancellationToken.None).ConfigureAwait(false);
+            };
 
-            // assert
+            // assert for connection status
             connectionStatusInfo.Status.Should().Be(ConnectionStatus.Disconnected);
             connectionStatusInfo.ChangeReason.Should().Be(ConnectionStatusChangeReason.DeviceDisabled);
+            // assert for exception
+            var error = await act.Should().ThrowAsync<IotHubClientException>();
+            error.And.StatusCode.Should().Be(IotHubStatusCode.DeviceNotFound);
```

Comment:
Does this work?

```suggestion
            await act.Should().ThrowAsync<IotHubClientException>()
                .And.StatusCode.Should().Be(IotHubStatusCode.DeviceNotFound);
```

Reply:
No, I tested this previously but somehow it didn't